### DISCUSSION TITLE
How to get access token from social login

### DISCUSSION THREAD

**Discussion by dellwatson** [reown-com/appkit#3142] [Q&A]
Hello is it possible to get the social login account detail? the email, and perhaps access token. 
it would be good if there's option to add the auth0, so i can get those access token. 

because im also using api from twitter and google, so i need the access token and it's not ideal to connect social login twice

**Comment by mdqst** [comment]
WalletConnect does not directly provide access to social login account details, such as email or access tokens. However, you can integrate Auth0 to manage authentication and retrieve access tokens. This way, you can connect your social logins through Auth0 and avoid multiple connections for services. For more details on implementing this, you can check the WalletConnect documentation or the Auth0 integration guide.

**Reply by KazanderDad** [reply]
Where can I find this documentation? Can you provide any specific examples on how to get access to the email from the social login? Can it be achieved with the free tier from Auth0, or do we need the the premium features?

**Comment by arein** [comment]
@dellwatson @mdqst we're currently building something towards this. Can you DM me on Twitter? I'd love to get in touch and confirm whether what we're building meets your needs https://x.com/cyberdrk

**Reply by KazanderDad** [reply]
Your DM on Twitter isn't open :-(

**Reply by arein** [reply]
ups. fixed

**Comment by stefda** [comment]
@arein, I'm in a similar situation. I'm aware that AppKit now exposes the email of the authed user, but if I want to verify that they're who they're claiming to be on our backend, I need some sort of access token. AFAIK you don't currently expose access token that's part of the Google oauth flow, for example.

How do you suggest we tackle the following scenario:

1. User logs in using their Google creds via AppKit socials login
2. Frontend calls backend with the authed credentials for account access
3. Backend verifies credentials and exchanges them for a bearer (JWT, we're using supabase)
4. Frontend uses JWT for subsequent data retrieval from the backend

I'm finding the docs hard to navigate atm so any pointers in the right direction will be appreciated!

**Reply by arein** [reply]
hi - we'd finalizing requirements for this. Can you DM me on Telegram so we can build this with you? `cyberdrk`

**Comment by neverknowerdev** [comment]
hey! any updates on this? Really need that 🙂
